src/arch/Sound/RageSoundDriver_WDMKS.cpp

```diff
@@ -98,13 +98,12 @@ struct WinWdmFilter
 
 	~WinWdmFilter()
 	{
-		for( size_t i = 0; i < m_apPins.size(); ++i )
-			delete m_apPins[i];
+		m_apPins.clear();
 		if( m_hHandle )
 			CloseHandle( m_hHandle );
 	}
 
-	WinWdmPin *CreatePin( unsigned long iPinId, RString &sError );
+	std::shared_ptr<WinWdmPin> CreatePin( unsigned long iPinId, RString &sError );

```

Comment:
Should these be unique_ptrs instead of shared_ptrs? (I haven't read through this code super thoroughly yet fwiw)

Reply:
To be fair, I didn't attempt to get it working with `unique_ptr`. The returned value gets added to a `vector`, so I assume I'll need some `std::move` logic at the very least. I'll give it shot and see if it can get it working with `unique_ptr`.